### DISCUSSION TITLE
Migrating from the HACS Integration to the HA Integration

### DISCUSSION THREAD

**Discussion by Slim1399** [music-assistant/support#3259] [Q&A] [answered]
I followed the instructions and yet a "Warning" message surfaces "mass: move_integration_to_ha_core".

what does it mean?
![Capture d'écran 2024-12-05 041646](https://github.com/user-attachments/assets/993e316f-5799-4200-bad4-3e8ff7c94da9)

**Comment by OzGav** [comment]
You may not have removed the HACS integration as that is what creates that message. If you have definitely removed it (you can check the actions in the developer tools and see if any mass. Actions are there) then you can ignore this

**Reply by Slim1399** [reply]
I ignored the message then restarted and nothing, everything was back to normal a priori !

**Comment by the-mentor** [comment]
@OzGav I have the same issue and removing the HACS integration doesnt remove this message :(

**Reply by OzGav** [reply]
Just click on ignore then?

**Reply by the-mentor** [reply]
Doesn't it mean it didn't do something properly?

**Reply by OzGav** [reply]
No it is telling you that you need to do something. It is a repair which is just advisory

**Reply by the-mentor** [reply]
In my experience when you get a repair and you complete what it tells you to do it automatically goes away

**Reply by OzGav** [reply]
Ok

**Reply by dougmaitelli** [reply]
I assume this is a problem on how home assistant handles repairs and not from the music assistant integration itself.

Yes, we can just ignore the repair, but it will forever show in the ignore repairs list and that bothers me. Ideally this should be marked as resolved automatically.

I think that the hacs integration created the repair action and when that gets removed there is no one to actually mark it as resolved and home assistant does not remove repairs from removed integrations (and this is the issue imo)

**Comment by PockyBum522** [comment]
I also have this after I definitely removed music-assistant from HACS and watched it disappear from the list and rebooted HA. I even accidentally removed the server addin from HA, so there should be nothing music assistant in my system at the moment, and I'm still getting the message. Not sure if it just sticks forever once the message gets posted, but feels weird that there should be nothing left over to generate this message but that the message still shows up.

**Reply by OzGav** [reply]
@jozefKruszynski

**Reply by jozefKruszynski** [reply]
Hmm, I guess we need to fix that pronto.

**Reply by dunxd** [reply]
@jozefKruszynski Until it is fixed it would nice to include some guidance in the documentation at https://music-assistant.io/integration/installation/#migrating-from-the-hacs-integration-to-the-ha-integration - at the moment the persistence of the warning message suggests that not all steps have been completed properly, and it is not clear whether this is the case or not.  Most people seeing the warning are going to click on the Learn More link so that should note this issue.

**Comment by ministryofsillywalks** [comment]
Is it planned that the official integration will catch up to the hacs features?
I use the search action very often and official doesn't Support that. 
(Found that out after gleefully migrating to official and then wondering why all my automations stopped working... Read the manual I should... :D

**Reply by OzGav** [reply]
Yes. In about a month.

**Comment by OzGav** [comment, marked as answer]
This should be fixed with HACS Integration version 2024.12.1

**Comment by jozefKruszynski** [comment]
Issue fixed in new release. The fix is auto applied when removing the Music Assistant (CUSTOM) integration.

If you have already removed the HACS integration and/or ignored the issue, you will need to reinstall this new version, set it up and then simply remove it, and the issue should be resolved.

https://github.com/music-assistant/hass-music-assistant/releases/tag/2024.12.1

**Reply by dunxd** [reply]
After I had removed it already I was getting the error and ignored it.  Here are the steps I took to resolve it:

1. Settings > System > Repairs > vertical dot menu > Show ignored errors - then I unignored the error
2. HACS > Music Assistant > Download
3. Restart Home Assistant
4. Add the integration
5. Settings > Devices & Integrations > Music Assistant (Custom)  > three dot menu > Delete
6. Restart Home Assistant

Now the message is gone from settings. Done.

**Reply by jozefKruszynski** [reply]
You shouldn't even need to unignore, but thank you for providing a more detailed explanation than me. We appreciate it very much

**Reply by fslef** [reply]
I'm still facing the issue. How to procees as the repo is now public archive ? is there any way to clean it up manyally to suppress the warning message ?

**Reply by OzGav** [reply]
Ok we have unarchived it for you. Be quick as it will be archived again soon.

**Reply by fslef** [reply]
`Downloading music-assistant/hass-music-assistant with version 7203004 failed with (Could not download, see log for details)`

This might be due to the lack of recent releases in the repository.

I’m glad the repository was unarchived, but since not everyone updates their Home Assistant instance monthly, others might encounter similar issues when trying to update later. A more long-term solution could be beneficial for them

**Reply by Kieselmania** [reply]
Hi, got the same error, too.
These are my steps to bypass the problem:

I followed the steps of @dunxd!
Instead of downloading in HACS in step 2, perform the following steps:

1. go to [https://github.com/music-assistant/hass-music-assistant](https://github.com/music-assistant/hass-music-assistant)
2. download Zip 
![image](https://github.com/user-attachments/assets/2eebe043-5c84-4671-806b-d93abf1ffbdb)
4. Unpack Zip on your computer
5. copy the folder "mass"
6. paste it into folder "custom_components" of Home Assistant

Continue with step 4 @dunxd and don't forget to delete the folder "mass".

Worked perfect! :-)

**Reply by dunxd** [reply]
These steps really need to be added to the documentation linked to from the warning message, particularly now the HACS version is to be archived.

**Reply by fslef** [reply]
In step 1, your link is dead. 

Anyway, I've dowloaded the repo archived, expanded it, and copied the mass folder into the custom_components folder of my home assistant. 

I rebooted my Home Assistant, but when I tried to add the `Home Assistant (Custom)` Integration it fails to connect the music assistant instance. (so the integration configuration is canceled)

Difficult to troubleshoot more into details as the former documentation that explain the manual installation steps is no longer available.

I would suggest the team should work on a more reliable fix.

**Reply by fslef** [reply]
> These steps really need to be added to the documentation linked to from the warning message, particularly now the HACS version is to be archived.

I don't see yhe message where you are detailing those steps

**Reply by jozefKruszynski** [reply]
First of all, sorry that we created this problem, we made a mistake.
Having said that, the fix was available for about 1 month. We assumed this was ample time for people to apply the fix.
We will try to create a solution that fixes this issue. Please bear in mind that the easiest way to fix this is to directly manipulate the underlying HA config files, however releasing something that manipulates any config files outside of the HA apis is not something we will do.
As said, we're considering options, but you have to be patient as not only is this a free and open source project that people volunteer their time to, it is also still holiday season for many.

**Reply by dunxd** [reply]
> I don't see yhe message where you are detailing those steps

Rolled up - https://github.com/orgs/music-assistant/discussions/3259#discussioncomment-11494844 plus the suggestions of @Kieselmania in 
https://github.com/orgs/music-assistant/discussions/3259#discussioncomment-11736634

**Reply by xvlw** [reply]
What are the best/current steps to remove this from my repair warnings.

I went to the documentation area where I used to see the steps and where the warning points me when I click "Learn More" and they are no longer on that page : https://www.music-assistant.io/integration/installation/#migrating-from-the-hacs-integration-to-the-ha-integration 

Appreciate your help in getting this removed from my system.

**Reply by VACInc** [reply]
What is the reason for not keeping this available for anyone to do at any time? A little confusion? I think it would still be very obvious when they go to install it via HACS that is DEPRECATED

**Reply by OzGav** [reply]
@xvlw Just follow the two links in the post immediately above yours

**Reply by OzGav** [reply]
@VACIndustries because some people don't behave as you would expect and that causes us work. We expect that people would have migrated months ago as indicated by the repair and if they didn't then they knew what they were doing. As per the posts above there remains a way to remove the repair notification manually.

**Reply by xvlw** [reply]
@OzGav  I sincerely apologize that I did not have time to tackle the mistake your team made within the time frame you defined....  What work have I caused you by trying to resolve an issue with Music Assistant that I did not cause in the first place? 

If you want to talk about making more work for someone, how about this exact action I am forced to resolve. 

I still do not understand how my inability to resolve this issue (that I did not create) within the time frame you defined has any bearing on your choice to remove the instructions from the support documentation. The same documentation that is linked in the REPAIR card.

**Reply by xvlw** [reply]
Separately for any humans who are still stuck with this error. If you follow the above instructions you may get hit with a "Failed to Connect" after trying to complete Step 4 (add the integration)

This is because you ALSO have to go to the Add On store and download and run the Music Assistant Server...

So prior to Step 4. Navigate to Add On Store, download and run Music Assistant Server. Once its running resume Step 4 and add the Integration.

This is ridiculous. Do better guys..

**Reply by marcelveldt** [reply]
I suggest you to dim down your tone. You are aware this is free software created by people in their free time for you ?
There is no huge dev team, there is test team, there is no quality control department. YOU are the test team, in exchange for the fact you can use this for free. Try to respect the people helps a lot in getting support.

**Reply by VACInc** [reply]
Whereas I don't agree with the tone -- I agree in how this was not executed well.  It makes more sense for easy removal an clear instructions to point to for the likely fewer that will somehow end up manually installing the HACS integration even though it blatantly states DEPRECATED everywhere.

At this point the amount of time spent here discussing this would have already been avoided had it been done the way suggested above.

**Reply by xvlw** [reply]
@marcelveldt  The tone was a reaction of a member of your team stating the below that we are causing your team more work. 

> because some people don't behave as you would expect **and that causes us work**. We expect that people would have migrated months ago as indicated by the repair and if they didn't then they knew what they were doing

We understand mistakes happen, however when we are seeking guidance on how to clear the error after the instructions were removed from the documentation, this is also going to generate the "extra work" that is being complained about by the MA team member.

While your team is volunteer based, that doesn't excuse a member from coming in here with their own tone when users are trying to seek help with the initial error and the removal of the instructions on how to clear the error.

I think this topic has been beaten to death, we as users have made our point pretty clear, it's up to the MA team to address it if they feel they want to.

**Comment by FARKIr** [comment]
![image](https://github.com/user-attachments/assets/7621fd41-c3d0-4eb6-a836-f9075f82b187)
![image](https://github.com/user-attachments/assets/e5c5f747-a2fd-4854-a87e-a8e4ece7eae0)
![image](https://github.com/user-attachments/assets/0114ab4b-d327-4d8c-af29-78d47c837430)
![image](https://github.com/user-attachments/assets/8b1c15f7-0336-430f-8f5e-84a3fb3885e3)
![image](https://github.com/user-attachments/assets/7a8a205c-1d09-420e-b460-84879ccb8199)

**Comment by FARKIr** [comment]
![image](https://github.com/user-attachments/assets/b2dd82e5-f70f-448b-a9a8-b29462a1752b)
![image](https://github.com/user-attachments/assets/73297595-f835-4486-a49a-93f32530fb84)

**Reply by FARKIr** [reply]
a

**Comment by MankiniChykan** [comment]
Can't download fix
![Screenshot_20250308-204749](https://github.com/user-attachments/assets/714f13a0-8740-406e-a139-474c6d65cebe)

**Reply by OzGav** [reply]
Try manually as per the note here https://www.music-assistant.io/integration/migrate/

**Reply by MankiniChykan** [reply]
I tried that still won't

**Reply by OzGav** [reply]
Which part didn’t work

**Reply by MankiniChykan** [reply]
> Which part didn’t work

I ended up getting it working. I thought I would share some more concise steps for luddites like me.

**Copy the whole repo over to your ` /config/custom_components` folder, from:**
**https://github.com/music-assistant/hass-music-assistant/tree/main/custom_components/mass**
![image](https://github.com/user-attachments/assets/11917070-83ca-4900-b87c-e1a02959ba9a)

**I then  added the (CUSTOM) component from the `Integrations` Page in `Devices and Services`**

- ![image](https://github.com/user-attachments/assets/87bfd8e0-e19c-4e7e-832f-a99fec3c2fd3)
- ![image](https://github.com/user-attachments/assets/37e5b4e3-2ba7-4291-a967-ae544ee7f1f7)
- ![image](https://github.com/user-attachments/assets/84e3e514-2ead-4a41-9545-6a038f777df1)
- ![image](https://github.com/user-attachments/assets/04309471-cfc6-41e2-a4bb-ff53d94f21dc)

**Then I deleted the Custom component**

- ![image](https://github.com/user-attachments/assets/ad88543c-0e17-447d-9507-ace564feca94)
- ![image](https://github.com/user-attachments/assets/2fb254a2-19ed-4a4a-a263-df84d93faff6)

**Now FINALLY the annoying notification has disappeared.**

- ![image](https://github.com/user-attachments/assets/408e0013-cb72-4701-9f5e-7444aeba1b69)
- ![image](https://github.com/user-attachments/assets/a59fe403-4862-4552-9418-5b510580e1f5)

**Reply by sgebhardt** [reply]
thanks, finally the ignored error is gone.

**Reply by MankiniChykan** [reply]
@sgebhardt I am pleased, took me a while to figure out, hence posting the detailed steps to help others.

**Reply by salopette** [reply]
Vielen Dank! Endlich ist der Fehler weg!

**Reply by wal-and** [reply]
Thank you very much, it helped, the problem is gone 
:)

**Reply by MankiniChykan** [reply]
You are welcome

**Reply by wrettro** [reply]
Hi, I'm currently getting this error but trying @MankiniChykan's fix doesn't seem to work at the moment. When trying to add the custom integration it just gives me an invalid version error so I can't complete it. Am I SOL or is there still some way to get rid of this error?

**Reply by OzGav** [reply]
You will have to go into the HA config files in `.storage` and remove reference to it in there

**Reply by wrettro** [reply]
By any chance do you happen to know which files?

**Reply by OzGav** [reply]
I think core.config_entries. Be careful as if you corrupt that file your HA probably wont start.

**Reply by qkevinto** [reply]
@wrettro did you ever get around to fixing this? I attempted @OzGav's suggestion but the only reference I could find in `core.config_entries` was for the new Music Assistant integration which I deleted and it just reseted my new integration, the "mass: move_integration_to_ha_core" is still present.

**Reply by qkevinto** [reply]
Never mind, found a manual fix, documented it here: https://github.com/orgs/music-assistant/discussions/3259#discussioncomment-16017967

**Comment by qkevinto** [comment]
For anyone still running into this issue, or perhaps dug into their list of ignored repairs and realised this entry is still there from >2 years and you have long removed the old integration, the fix is pretty straight forward.
1. Make a backup of HA just in case
1. Edit `/config/.storage/repairs.issue_registry` (use whatever tool you wish to do so, I used [Advanced SSH & Web Terminal](https://github.com/hassio-addons/app-ssh)), remove this entry:
    ```
    {
      "created": "2024-12-05T02:12:11.918410+00:00",
      "dismissed_version": null,
      "domain": "mass",
      "is_persistent": true,
      "issue_id": "move_integration_to_ha_coremass",
      "breaks_in_ha_version": "2024.12.0",
      "data": null,
      "is_fixable": false,
      "issue_domain": "mass",
      "learn_more_url": "https://music-assistant.io/integration/installation/#migrating-from-the-hacs-integration-to-the-ha-integration",
      "severity": "warning",
      "translation_key": "move_integration_to_ha_core",
      "translation_placeholders": {
        "domain": "mass",
        "integration_title": "mass"
      }
    },
    ```
1. Reboot
1. Repair warning should be gone for good
